version: "3"

services:
  hub-db:
    image: postgres:9.5@sha256:ee604a9025ec864da512736f765ae714112a77ffc39d762c76b67ca0f8597e9e
    container_name: jupyterhub-db
    restart: always
    env_file:
      - shared.env
    volumes:
      - "db:/var/lib/postgresql/data"
    networks:
      - internal

  hub:
    depends_on:
      - hub-db
    image: materialdigital/pyiron-jhub:2021-06-07@sha256:4a3af95fc60d5ad0ac6331721a48271b4f2a4dc3ef2d378c0c06a60a2f3a0831
    restart: always
    container_name: jupyterhub
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:rw"
      - "data:/data"
      - "user_auth_hook.py:user_auth_hook.py:ro"
      - "${HOST_CERT_PATH}:/srv/jupyterhub/secrets/jupyterhub.crt:ro"
      - "${HOST_KEY_PATH}:/srv/jupyterhub/secrets/jupyterhub.key:ro"
    ports:
      - "443:443"
    links:
      - hub-db
    env_file:
      - shared.env
      - pyiron.env
    environment:
      POSTGRES_HOST: hub-db
    command: >
      jupyterhub -f /srv/jupyterhub/jupyterhub_config.py
    networks:
      - internal
      - pmd-server_nginx-net

volumes:
  data:
      name: jupyterhub-data
      driver: local
  db:
      name: jupyterhub-db-data
      driver: local

networks:
  pmd-server_nginx-net:
    external: true
    name: ${NGINX_NETWORK}
  internal:
    name: jupyterhub-network
    driver: bridge

